### ISSUE TITLE
[Feature Request] Recursive Variables / Variables referencing variables

### ISSUE THREAD

**Issue by tsteckenborn** [usebruno/bruno#956]
As of now variables referencing other variables is not supported.

The current implementation of the user interface falsely suggests that this might be possible, yet executing requests ignores variables within the variable.

![Recursive Variables in Bruno Environment 1](https://github.com/usebruno/bruno/assets/50519619/42678fcc-a157-4fbb-b927-70348e1b2dbb)
![Recursive Variables in Bruno Environment 2](https://github.com/usebruno/bruno/assets/50519619/b8f4330c-0ddf-4994-91c4-868fb0697c58)

This functionality is supported within [Postman](https://community.postman.com/t/variables-within-variables/4082/2) and [Insomnia](https://docs.insomnia.rest/insomnia/environment-variables#recursive-variables).

It would be nice for Bruno to support this functionality as well. If this is not planned it should be thought about User Interface modifications so that there's no false suggestion of that working.

**Comment by adarshlilha** [comment]
this is possible to implement by replacing the `{{var}}` with the actual value by searching `formik.values` and replacing the same.

I tried this, but setting value of the field is changing the value of `{{var}}` in input box itself
Any pointers on how a var reference is converted to it's value in URL bar would be helpful as reference

should this logic live inside `onSubmit` of the form - thoughts @helloanoop ?

**Comment by Primajin** [comment]
The poor-man's solution I found so far was to go my `Collection`'s root -> `Script` and do something like this:

```js
const scheme = bru.getEnvVar('scheme');
const host = bru.getEnvVar('host');
const base_path = bru.getEnvVar('base_path');

bru.setEnvVar('base_url',`${scheme}://${host}${base_path}`);
```

<img width="641" alt="Screenshot 2024-01-03 at 16 18 33" src="https://github.com/usebruno/bruno/assets/1742115/59b6e574-8a40-464d-8479-5245f188d2a0">

**Comment by Nahuel92** [comment]
Please bump, this is a very handy feature that would be awesome to have on Bruno.

**Comment by cciudadtorrejon** [comment]
Bumping, we are using Terraform to setup dynamic environments and this feature will be so great for us.

**Comment by crimsonvspurple** [comment]
One more request to prioritize this.

**Comment by Nahuel92** [comment]
Hi @helloanoop, is there any chance to have this merged anytime soon?
Thanks!!

**Comment by SanampreetSingh** [comment]
Hi @helloanoop @sanjai0py - Can we merge this please? It's long pending feature request.
There are multiple issues/requests created and parallel fixes made for this already. Thanks!

**Comment by helloanoop** [comment]
We are prioritising this. Will share more updates later this week.

**Comment by sanjai0py** [comment]
Closing this as the [PR](https://github.com/usebruno/bruno/pull/2234) that implemented the feature was recently merged and released in [v1.20.0](https://github.com/usebruno/bruno/releases/tag/v1.20.0).
